### ISSUE TITLE
Category: Introduce new model and MVC

### ISSUE THREAD

**Issue by ybakos** [osu-cascades/hdffa-directory#38]
Attributes:

- `name`. Unique. No Null.

**Comment by ybakos** [comment]
We'll eventually associate this with Providers, but I'm not sure if they'll want a 1:M or M:N (whether providers will just have one category or be able to be in multiple categories).

Regardless, creating the categories crud will enable them to specify the category names.

**Comment by ybakos** [comment]
Let's talk before you tackle this one.

**Comment by ybakos** [comment]
Looks like this is in progress, great!

When I use the rails scaffold to get started, I usually try to...

1. Run the migration.
2. Add model validations and unit tests.
3. Re-do #1 and/or #2 if I didn't quite get it right, or until I'm happy.
4. Play with the feature to make sure I have my head around it. (eg fire up the server, visit /categories and do some creating, editing, deleting, etc. This helps me see what the scaffold generated.
5. Then move on to the controller and/or view layer.

In other words, I try to get the model layer solid before worrying about anything else.
